ultravox/inference/infer.py

```diff
@@ -38,6 +38,29 @@ def __init__(
             None
         )
 
+        assert self.tokenizer.padding_side == "left"
+
+    def batch_infer(
+        self,
+        samples: List[datasets.VoiceSample],
+        max_tokens: Optional[int] = None,
+        temperature: Optional[float] = None,
+    ) -> base.InferenceGenerator:
+        inputs = [self._dataproc(s, batch=True) for s in samples]
+        data_collator = datasets.DataCollatorForSeq2SeqWithAudio(
+            tokenizer=self.tokenizer,
+            include_alt_fields=False,
```

Comment:
should the `data_collator` be a member of the inference instance, given that it is needed for each batch?

Reply:
Yeah, would be nice to not reinitalize everytime we do a batch.